File: tests/python/cases/case_077_route_session.py
"""
Task case 077: route session payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_077_route_session(input_data):
    # Task: route session records and prepare transport-ready payload.
    task_name = 'route_session'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'session',
        'stage': 'route',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is appended to RETAINED_AUDIT, outliving function scope.
    RETAINED_AUDIT.append(payload)
    return 'ok'

File: tests/nodejs/cases/case_077_route_session.js
/**
 * Task case 077: route_session
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case077RouteSession(input) {
  // Task: route session records and prepare transport-ready payload.
  const taskName = 'route_session';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'session',
    stage: 'route',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is appended to retained audit state, outliving function scope.
  retainedAudit.push(payload);
  return 'ok';
}

module.exports = {
  case077RouteSession,
};
